appjs/app_init.js

```diff
@@ -38,6 +38,43 @@
         this.name = "AddonModHsuforumViewLinkHandler";
     }
 
+    function AddonModHsuforumMarkNotificationRead(urlHash) {
+        let site = t.CoreSitesProvider.getCurrentSite();
+        let toUser = site.getUserId();
+
+        const data = {
+            useridto: toUser,
+            useridfrom: 0,
+            type: 'notifications',
+            read: 0,
+            newestfirst: 1,
+            limitfrom: 0,
+            limitnum: 0
+        };
+
+        // Check if there is an unread notification that matches urlHash. Below todo will fix this section.
+        site.read('core_message_get_messages', data).then((response) => {
+            if (response.messages) {
+                let unreadNotification = response.messages.filter(notification => notification.contexturl.includes(urlHash));
+
+                // Mark notification as read if found
+                if (unreadNotification.length) {
+                    let params = {
+                        messageid: unreadNotification[0].id, 
+                        userid: toUser,
+                        markallnotifications: 0, 
+                    };
+                    Promise.resolve(site.write('theme_legend_custom_notifications', params)).catch((err) => {
+                        // TODO refactor above function to include notification id in urlcontext and change return type to json
+                        // This will throw an error at this point due to the moodle mobile post function only accepts json as 
+                        // a return type.
+                        console.log(err);
```

Comment:
This log okay? If so, could it maybe be console.error().

Reply:
The idea behind this is so that the promise error is handled gracefully. The promise returns an error because the web service return type is not in the excepted XHR return types defined by the moodle post service.

The request still completes but gives that warning. Without catching it, it gives a nasty big error.